src/Chunk.ts

```diff
@@ -1021,13 +1033,21 @@ export default class Chunk {
 
 	private getChunkExportDeclarations(): ChunkExports {
 		const exports: ChunkExports = [];
-		for (const exportName of this.getExportNames()) {
+		const exportsNames = new Set<string>();
+		for (let exportName of this.getExportNames()) {
 			if (exportName[0] === '*') continue;
 
-			const variable = this.exportNames[exportName];
+			let variable = this.exportNames[exportName];
 			const module = variable.module;
 
 			if (module && module.chunk !== this) continue;
+			const isEntryPoint = !!(this.facadeModule && this.facadeModule.isUserDefinedEntryPoint);
```

Comment:
The `!!` seems superfluous. More importantly, `isEntryPoint` seems quite misleading. There seem to be quite a few definitions of entry point floating around and it is quite unclear why this specific case needs a different logic while others do not. Will it work correctly for dynamic imports, which are non-user-defined entry points?

More importantly, I do not think it quite hits the spot. I just tried the following test case (that you might consider to add), which has *interesting*  results:

```js
// main.js - entry point
import { foo } from './main2.js';
console.log(foo);

// main2.js - entry point with synthetic exports
export {bar as default} from './dep.js';

// dep.js - synthetic exports
export default {bar: {foo: 'works'}};
```

What I get is
- two default exports in `generated-main2.js`
- an invalid import in `main.js`

Reply:
This is the trickiest problem, ideally we would love to pass the default export (object) around, so live references work, however, when exporting in "entry-points" the synthetic exports needs to become real named exports.

Maybe any re-export should always create an auxiliar variable in ESM?

Reply:
I am not sure entry-points can have synthetic named exports though, seems pretty impossible thing to do. Rollup needs to emit actual valid ESM, and ESM does not have such a thing

Reply:
They an if you use a CJS entry point–which is a thing that people do. Of course such an entry point cannot expose arbitrary exports to the outside, but arbitrary internal imports from this file should still work.

> Maybe any re-export should always create an auxiliar variable in ESM?
I would hope this can be avoided for internal imports.

Looking at `generateInternalExports`, you see that the presence of a `facadeModule` is the sole indicator if a file is "internal" with auto-generated exports with arbitrary names or "external" following exactly the naming given in a facadeModule.